lib/pbench/server/database/alembic/versions/9cc638cb865c_operationname.py

```diff
@@ -0,0 +1,36 @@
+"""Correct operationname ENUM
+
+Alembic does not check or autogenerate ENUM value changes, so this will
+upgrade the original operationname revision to add UPDATE.
+
+Revision ID: 9cc638cb865c
+Revises: 9df060db17de
+Create Date: 2023-03-03 14:32:16.955897
+
+"""
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "9cc638cb865c"
+down_revision = "9df060db17de"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # NOTE: the 'BACKUP' and 'UNPACK' operationname ENUM values are now
+    # obsolete. We don't attempt to remove them because dataset_operations
+    # rows with these values record the previous operation of the server.
+    # Instead we simply add the new 'UPDATE' value. Having the obsolete
+    # values defined in the ENUM isn't a problem, and removing them isn't
+    # worth the complication.
+    with op.get_context().autocommit_block():
+        op.execute("ALTER TYPE operationname ADD VALUE 'UPDATE' BEFORE 'UPLOAD'")
+    # ### end Alembic commands ###
+
+
+def downgrade() -> None:
+    # Downgrade is problematic, and won't be attempted. Having unused ENUM
+    # values defined shouldn't represent a problem.
+    pass
```

Comment:
I always wonder whether we should ever try to downgrade. Meaning we can always do the incremental upgrade and accomplish the same effect as the `downgrade` if we ever want to downgrade the DB revision. The upside would be that it's easier to follow what's the current schema in our DB and what's changed from the last revision. Thoughts?

Reply:
Normally alembic autogenerates downgrades as well as upgrades. In practice, yeah, I doubt that we ever want to "support" this ... it'd be too complicated to confirm that it really works.

Reply:
I would approach downgrade with great trepidation.  (E.g., I would seriously consider the alternative of restoring the database from backup and re-adding any missing datasets...failing that, I would probably advocate for crafting an "upgrade" to the database to get it back to its previous schema rather than actually running the downgrade, although, in the end, the only difference between the two should be the resulting Alembic hash....)

Reply:
Yeah, if we follow that then we wouldn't have to keep the old revision files in our repository.

Reply:
The "chain" of revision files represent incremental upgrades: we can't drop any because the current database state is a journey from the original revision through the latest. Combining multiple revisions would be a mess, and would preclude updates from the interim versions, so we wouldn't want to do that. Whether or not we support downgrade is a completely unrelated issue.